src/modules/validation.ts

```diff
@@ -280,13 +280,16 @@ export interface RuleJSON extends z.infer<typeof ruleValidator> { }
  * @returns Either the parsed RuleJSON object or an error.
  */
 export const validateRuleJSON = (
-  rule: string
+  rule: string,
+  reversed: boolean = false
```

Comment:
Thoughts on having a distinct validator for rule and and reversed rule, i think the usage of each is distinct in the code.

Reply:
With what you currently have there is a type mismatch is in what is declared by the function and what is being returned and the error message is less clear.
All of the validate{TYPE} functions are very similar, all performing the same routine, validate, check result, then format and return error of return successful result. I think they are all good candidates to be refactored with a single configurable function that can return a validate function.